### ISSUE TITLE
Support for lazy emit.

### ISSUE THREAD

**Issue by aindlq** [webpack/webpack-dev-middleware#141]
Is it possible to emit bundles lazily, only when user requests one (e.g in combination with webpack-dev-server)?

I have many bundles and some of them are quite huge and not all developers need to touch them, but still webpacks compiles/bundles them all the time even when bundle was not requested by dev-server. So I'm wondering if it is possible to have something more lazy than the current lazy option which just delay bundling. If it is not currently possible do you think it feasible to implement, if so any pointers are welcome!

**Comment by SpaceK33z** [comment]
> if it is possible to have something more lazy than the current lazy option which just delay bundling

What do you mean with this? That is not true, if you use the `lazy: true` option, it will only build when you actually request the bundle.

**Comment by aindlq** [comment]
Maybe it is my misunderstanding, but for me it always emits everything. Here is small example with 2 entry points and lazy dev server - https://github.com/aindlq/react-hot-boilerplate/tree/lazy-bundling. As soon as I open index.html in the browser it says that second bundle has been also emitted:

```
Listening at http://localhost:3000/
webpack: wait until bundle finished: /static/app-bundle.js
Hash: c822c376723b487622ce
Version: webpack 2.1.0-beta.25
Time: 1374ms
         Asset    Size  Chunks             Chunk Names
app2-bundle.js  719 kB       0  [emitted]  app2
 app-bundle.js  719 kB       1  [emitted]  app
webpack: bundle is now VALID.
```

Also if I then do some change to second bundle and again reload html page with the first one it bundles second one as well:

```
webpack: wait until bundle finished: /static/app-bundle.js
Hash: 10c6e5e539bb18b527cf
Version: webpack 2.1.0-beta.25
Time: 189ms
         Asset    Size  Chunks             Chunk Names
app2-bundle.js  719 kB       0  [emitted]  app2
 app-bundle.js  719 kB       1             app
webpack: bundle is now VALID.
```

Any ideas on what am I doing wrong?

**Comment by aindlq** [comment]
btw I have the same behaviour with 2.1.0-beta.25 as well as 1.13.1.

**Comment by SpaceK33z** [comment]
Ahh, you're working with multiple entry points. That changes things. It is not possible to only compile the bundle for one entry point, and this would not be easy to change.

Have you looked into using the `filename` option? This allows you to specify exactly on what request the bundle should recompile.

**Comment by aindlq** [comment]
Could you please explain in what case `lazy` works. It doesn't work for me with multiple entry points and it doesn't work with one entry point + code splitting with `require.ensure` as well.

**Comment by SpaceK33z** [comment]
`lazy` is meant to only recompile when you manually refresh the browser. It is not meant to enable partial recompiling for only one entry point. This feature would require the ability to do partial recompiling, which webpack doesn't have at the moment AFAIK.

**Comment by SpaceK33z** [comment]
If you have a huge bundle that takes long to recompile, I'd advise you to look into DLL plugin.

**Comment by ramasilveyra** [comment]
When https://github.com/webpack/webpack-dev-middleware/pull/151 arrives, it will be possible to support "lazy emit" (compile only what it's requested) with multi-compiler configuration?

**Comment by shellscape** [comment]
@ramasilveyra in theory, I believe so yes. #187 was merged, resolving #155. given the discussion and the mentioned PR, closing this one.

**Comment by shellscape** [comment]
@dmnd have a look at #187 (which is the fix of #155 that was abandoned). "the discussion" mentioned was this thread, plus those PRs, and the context in both. With #187 and the multiple compiler support, you should be able to duplicate partial compilation on some scale. But as @SpaceK33z mentioned, true partial compilation is not presently possible and would not be trivial (or even moderately inconsequential) to implement.

**Comment by dmnd** [comment]
Thanks @shellscape, I'll look into a multicompiler setup. This would be moving each of my entry points to their own compilation setup, right?
